aurora/aurora-war/src/main/java/org/silverpeas/looks/aurora/LookAuroraHelper.java

```diff
@@ -352,6 +351,18 @@ public NewsList getNews() {
 
   public NextEvents getNextEvents() {
     List<String> allowedComponentIds = asList(getAllowedComponentIds("home.events.appId"));
+    return getNextEvents(allowedComponentIds);
+  }
+
+  public NextEvents getNextEvents(List<String> allowedComponentIds) {
+    boolean includeToday = getSettings("home.events.today.include", true);
+    int nbDays = getSettings("home.events.maxDays", 2);
+    boolean fetchOnlyImportant = getSettings("home.events.importantOnly", true);
+    return getNextEvents(allowedComponentIds, includeToday, nbDays, fetchOnlyImportant);
+  }
+
+  public NextEvents getNextEvents(List<String> allowedComponentIds, boolean includeToday,
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 25 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS3776)